### PR TITLE
feat: create Markdown primitive

### PR DESCRIPTION
**Pull request by kimak**
## Checklist before requesting a review
- [x] Working on iOS
- [x] Working on Android
- [x] Integration tests added
- [x] Visual regressions screenshots are up to date
- [ ] Design validated

### CODE REVIEW

**Review by hcourthias — APPROVED** [review]

File: packages/streamline/src/primitives/markdown/markdown-link.tsx
Comment (kimak) [inline packages/streamline/src/primitives/markdown/markdown-link.tsx:55]: @judastheo I created a new MarkdownLink component and used it in CarTip
The goal is to avoid too much coupling with Markdown component. 

I wait for your approval on the dedicated commit before merging
https://github.com/GetLuko/streamline/pull/61/commits/e36bf6c99a9e71d28727fdb828e42f70f655e15b

File: packages/sandbox/e2e/android/screenshots/Markdown.png
Comment (kimak) [inline packages/sandbox/e2e/android/screenshots/Markdown.png:1]: @ianrobertlambert waiting your approval on Android before merging

File: packages/sandbox/e2e/ios/screenshots/Markdown.png
Comment (kimak) [inline packages/sandbox/e2e/ios/screenshots/Markdown.png:1]: @ianrobertlambert waiting your approval on iOS before merging

**Review by judastheo — APPROVED** [review]
@kimak Good compromise I think, all clear!

File: packages/sandbox/e2e/ios/screenshots/Markdown.png
Comment (kimak) [reply packages/sandbox/e2e/ios/screenshots/Markdown.png:1]: @ianrobertlambert I will merge it from a code point of view. Also because we already synced. But let me know if you have any feedback and I can iterate

**Comment by ianrobertlambert** [comment]
Hey @kimak - As we discussed in sync, thanks for all of this! After taking a closer look, I had some small feedback (hopefully 🫠).

- [ ] Could we standardize the indentation to be 48px from the margin for visual alignment as you move down the screen?
image

- [ ] It would be helpful to show how images with different aspect ratios, such as `16:9,` `1x1`, and `2x3`, can be scaled to demonstrate this to stakeholders.
image

- [ ] Can we please add a `8px` border radius and `4px` of padding to the left and right of the inline code.
image

- [ ] The type size for blockquotes seems slightly small; are we using `Title SM` here?

- [ ] On Android, the Bold and Italic Bold formatting looks like it's using the wrong typeface. Could it be a rendering issue for me?

- [ ] The numbers on the ordered list seem a bit small; are we using the `Body` typeface here?

- [ ] It would be easier to provide accurate feedback if we can see the examples wrap. I've made some quick adjustments to the markdown sandbox page, and it could help to see how the markdown is used in context between elements. WDYT? https://www.figma.com/file/tOJLNigXQXJqCSKgJFPZn5/App-Streamline-Lukomponents?node-id=5448%3A15280&t=XTDSSy8UfzS7PCu9-1

**Comment by ianrobertlambert** [comment]
Sorry, I think this feedback on Markdown was lost because it was merged. Should I create a new issue? cc @judastheo @kimak @hcourthias

**Comment by kimak** [comment]
thanks for pointing it @ianrobertlambert I missed it. I will take your feedback today

**Comment by kimak** [comment]
@ianrobertlambert here is my update:

> - [x] Could we standardize the indentation to be 48px from the margin for visual alignment as you move down the screen?

> - [x] It would be helpful to show how images with different aspect ratios, such as `16:9,` `1x1`, and `2x3`, can be scaled to demonstrate this to stakeholders.

I added it at the end so you should scroll with expo app to see them.

> - [ ] Can we please add a `8px` border radius and `4px` of padding to the left and right of the inline code.
image

It's not possible with the actual implementation of the lib because the text should wrap with other texts.
This markdown was not in Figma initially, should we keep or delete?

> - [x] The type size for blockquotes seems slightly small; are we using `Title SM` here?

Good catch! It's because blockquote should be wrap as title: `> # Title quote example` 
I updated the expo example. 

> - [ ] On Android, the Bold and Italic Bold formatting looks like it's using the wrong typeface. Could it be a rendering issue for me?

Seems good on my side. Could you check the Luko app version with Markdown? it's the same font, but module fonts can differ on expo.

> - [x] The numbers on the ordered list seem a bit small; are we using the `Body` typeface here?

Good catch. Fixed

> - [x] It would be easier to provide accurate feedback if we can see the examples wrap. I've made some quick adjustments to the markdown sandbox page, and it could help to see how the markdown is used in context between elements. WDYT? https://www.figma.com/file/tOJLNigXQXJqCSKgJFPZn5/App-Streamline-Lukomponents?node-id=5448%3A15280&t=XTDSSy8UfzS7


Also added when you scroll, example here: 

Screenshot 2023-06-13 at 18 24 15
Screenshot 2023-06-13 at 18 24 28
Screenshot 2023-06-13 at 18 24 42

**Pull request merged**
